table/block.cc

```diff
@@ -283,9 +294,33 @@ bool BlockIter::ParseNextKey() {
 
     value_ = Slice(p + non_shared, value_length);
     while (restart_index_ + 1 < num_restarts_ &&
-           GetRestartPoint(restart_index_ + 1) < current_) {
+           GetRestartPoint(restart_index_ + 1) <= current_) {
       ++restart_index_;
     }
+    if (value_delta_encoded_) {
+      assert(value_length == 0);
+      if (GetRestartPoint(restart_index_) == current_) {
+        const size_t kHandleMaxSize = 16;
+        auto v = Slice(p + non_shared, kHandleMaxSize);
+        BlockHandle handle;
+        Status decode_s = handle.DecodeFrom(&v);
+        decoded_value_ = handle;
+        auto& updated_value = v;
+        assert(decode_s.ok());
+        value_ = Slice(p + non_shared, updated_value.data() - value_.data());
+      } else {
+        const size_t kHandleMaxSize = 8;
+        auto v = Slice(p + non_shared, kHandleMaxSize);
```

Comment:
How do we make sure Slice is within the bytes of the block?

Reply:
The slice would exceed the block limits only in the case of corruption. A corruption could be detected at any point during the reading. Without delta encoding we had this opportunity to detect such corruption when reading the value. With delta encoding we would rely either on past checks (checksum) or on future checks on the code to detect a value that does not make sense.